pkg/controller/servicemesh/controlplane/controller.go

```diff
@@ -255,6 +255,11 @@ func (r *ReconcileControlPlane) Reconcile(request reconcile.Request) (reconcile.
 		reqLogger.V(1).Info("Adding finalizer", "finalizer", common.FinalizerName)
 		finalizers.Insert(common.FinalizerName)
 		instance.SetFinalizers(finalizers.List())
+		// set default version if necessary
+		if len(instance.Spec.Version) == 0 {
+			instance.Status.AppliedVersion = common.DefaultMaistraVersion
```

Comment:
Now we're only initializing `Status.AppliedVersion` if the `Spec.Version` is unset.